tests/test_context.py

```diff
@@ -153,9 +164,24 @@ def test_root_fs_override(self):
         with chainerio.open(hdfs_tmpfile, "r") as fp:
             self.assertEqual(fp.read(), hdfs_file_string)
 
-        conn.delete(hdfs_tmpfile)
+        self.assertEqual(chainerio.isdir(hdfs_tmpnested_dir), True)
+
+        self.assertEqual(sorted(list(chainerio.list(hdfs_tmpdir))),
+                         sorted(["tmpfile_hdfs", "nesteddir_hdfs"]))
+
+        self.assertEqual(chainerio.exists(hdfs_tmpfile), True)
+        self.assertEqual(chainerio.exists(hdfs_tmpfile_renamed), False)
+
+        chainerio.rename(hdfs_tmpfile, hdfs_tmpfile_renamed)
+        self.assertEqual(chainerio.exists(hdfs_tmpfile), False)
+        self.assertEqual(chainerio.exists(hdfs_tmpfile_renamed), True)
+
+        chainerio.remove(hdfs_tmpdir, recursive=True)
+        self.assertEqual(chainerio.exists(hdfs_tmpdir), False)
         conn.close()
```

Comment:
This kind of resource collection must be cleanly done with `try: .. finally: ...`

Reply:
resolved in a8fa100